Feature/dropdowns single selector

weberjavi opened the pull request:
## single selector

### Overview

This PR creates a single-select dropdown.
It uses [downshift](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#statechangetypes) library to ease accessibility functionality.

It has light and dark themes.

![image](https://user-images.githubusercontent.com/6906348/104732262-3c3a3000-573d-11eb-9c92-a0d044a2c3e0.png)
![image](https://user-images.githubusercontent.com/6906348/104732632-ce423880-573d-11eb-8726-7b043865ad92.png)




### Designs

[Figma prototype](https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=1523%3A1582)

### Testing instructions

`yarn && yarn storybook:dev`

mbarrenechea commented on app/components/dropdowns/dropdown-single-select/component.tsx:
```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { useSelect } from 'downshift';
+import ARROW_DOWN_SVG from 'svgs/ui/arrow-down.svg';
```
```suggestion
import ARROW_DOWN_SVG from 'svgs/ui/arrow-down.svg?sprite';
```

If you are not in a story from storybook, remember to put `?sprite`

mbarrenechea commented on app/components/dropdowns/dropdown-single-select/component.tsx:
```diff
@@ -0,0 +1,173 @@
+  disabled?: boolean;
+}
+
+export interface SingleSelectProps {
```
Let's try to sort the interfaces by type, or at least, by groups related.

mbarrenechea commented on app/components/dropdowns/dropdown-single-select/component.tsx:
```diff
@@ -0,0 +1,173 @@
+  disabled?: boolean;
+  prefix?: string;
+  placeholder?: string;
+  options?: Array<Option>;
```
```suggestion
  options?: Option[];
```

Is this the same? If it is, please change it because consistency.

mbarrenechea commented on app/components/dropdowns/dropdown-single-select/component.tsx:
```diff
@@ -0,0 +1,173 @@
+  disabled = false,
+  prefix,
+  placeholder,
+  options = [],
```
Same order as the interface

mbarrenechea commented on app/components/dropdowns/dropdown-single-select/component.tsx:
```diff
@@ -0,0 +1,173 @@
+  clearSelectionLabel = 'Clear selection',
+  className,
+}: SingleSelectProps) => {
+  const itemToString = (option) => (option ? option.label : '');
```
Do we need this function? it's the same as putting `option?label`. I don't really see the necessity

mbarrenechea commented on app/tailwind.config.js:
```diff
@@ -188,7 +188,10 @@ module.exports = {
-    fill: { current: 'currentColor' },
+    fill: (theme) => ({
+      current: 'currentColor',
+      primary: theme('colors.primary.500'),
```
WHY? You can change all the svg fill by changing the `text-{color}-{number}` class

weberjavi commented:
@mbarrenechea closing this PR in favor of https://github.com/Vizzuality/marxan-cloud/pull/30
I've implemented the changes suggested on the other PR and added the multi-select dropdown.
